### PR TITLE
Add CI workflow for PR checks

### PR DESCRIPTION
**Pull request by Advaitgaur004**
1. **Automated Checks**: _This workflow automates the build, reducing the chances of introducing errors into the main branch._

2. **Immediate Feedback**: Contributors will receive immediate feedback on their changes, allowing quicker iterations and improvements.
 
3. **Consistency**: Ensures that all contributions are built and tested in a consistent environment.

### CODE REVIEW

File: src2/main.c [hunk 1]
```diff
@@ -39,12 +40,41 @@ int main() {
...
+        }
+    }
+    //Normalize the input
+    X = (const float(*)[4])X_norm;
+
     // create model
     Model model;
     cten_begin_malloc(PoolId_Model);
-    model.weight_1 = Tensor_new((TensorShape){n_features, 32}, true);
+    model.weight_1 = nn_random_init((TensorShape){n_features, 32}, true);
```
Comment (blueloveTH) [inline src2/main.c:75]: Keeps `Tensor_new` here and make random init inplace.

Comment (Advaitgaur004) [reply src2/main.c:75]: okay will do that

**Comment by Advaitgaur004** [comment]
@blueloveTH glorot initialization is written in place inside [Tensor_new](https://github.com/pocketpy/cTensor/blob/02b89bb13881b1c64d4e0c1f97dd8accb79ae1e3/src/basic.c#L36C8-L36C18)

**Comment by blueloveTH** [comment]
inside? That is completely incorrect.

**Comment by blueloveTH** [comment]
Only submit one pr. This is for evaluation and it will not be merged.

**Comment by Numbers0689** [comment]
> Only submit one pr. This is for evaluation and it will not be merged.

@blueloveTH  Just to clarify, are you referring to the changes to make main.c work? I've got it running with backpropagation, and while the loss fluctuates slightly, it decreases overall. Also, I run into use-after-free issue when increasing the number of layers and I continue to debug both of them.

**Comment by blueloveTH** [comment]
@Numbers0689 Yes. Try to fix `main.c` as much as you can. I will review students' result to evaluate their knowledge of neural networks and math.

This is not a contribution pull request. So if you find something wrong, you can fix and debug it, but no need to submit pull request.
